File: apps/mobile-native/android/app/src/main/java/mx/nexara/mobile/nativeapp/ui/integra/video/IntegraVideoRoutes.kt
package mx.nexara.mobile.nativeapp.ui.integra.video

/**
 * Rutas del bloque de video, para que quien cablee `IntegraNavHost` no tenga que
 * inventarlas ni copiarlas a mano.
 *
 * Este fichero no navega: solo declara. `IntegraNavHost.kt` es propiedad de otro
 * turno y este paquete no lo toca.
 *
 * Cableado esperado (dentro del `NavHost` de INTEGRA):
 *
 * ```
 * composable(IntegraVideoRoutes.WALL) {
 *     IntegraVideoWallScreen(
 *         onOpenCamera = { id ->
 *             nav.navigate(IntegraVideoRoutes.cameraDetail(id)) { launchSingleTop = true }
 *         },
 *     )
 * }
 * composable(
 *     route = IntegraVideoRoutes.CAMERA_DETAIL,
 *     arguments = listOf(navArgument(IntegraVideoRoutes.ARG_CAMERA_ID) {
 *         type = NavType.StringType
 *     }),
 * ) { backStack ->
 *     IntegraCameraDetailScreen(
 *         cameraId = backStack.arguments
 *             ?.getString(IntegraVideoRoutes.ARG_CAMERA_ID).orEmpty(),
 *     )
 * }
 * ```
 */
object IntegraVideoRoutes {

    const val ARG_CAMERA_ID = "cameraId"

    /** Rejilla de cámaras. */
    const val WALL = "integra/video"

    /** Detalle de una cámara. */
    const val CAMERA_DETAIL = "integra/video/{$ARG_CAMERA_ID}"

    /**
     * Un `cameraIndexCode` de Hikvision es alfanumérico con guiones, pero puede
     * traer caracteres que rompan la ruta. Se codifica para que un id raro no
     * mande la navegación a ninguna parte.
     */
    fun cameraDetail(cameraId: String): String =
        "integra/video/" + java.net.URLEncoder.encode(cameraId, "UTF-8").replace("+", "%20")

    /** Clave de módulo que hay que dar de alta para que la pantalla sea visible. */
    const val MODULE_KEY = "integra-video"

    /** Título de la barra superior para la rejilla. */
    const val TITLE_WALL = "Cámaras"

    /** Título de la barra superior para el detalle. */
    const val TITLE_DETAIL = "Cámara"
}

File: apps/mobile-native/ios/NexaraApp/UI/Integra/IntegraVideoRoutes.swift
import Foundation

/// Contrato de cableado del bloque de video.
///
/// Este fichero no navega: solo declara. `IntegraRootView` / `ModuleRouter` son
/// propiedad de otro turno. Cableado esperado vía `integraAdvancedDestination`.
enum IntegraVideoRoutes {
    static let argCameraId = "cameraId"

    /// Rejilla de cámaras (fotogramas JPEG, no MSE).
    static let wall = "integra/video"

    /// Detalle de una cámara.
    static let cameraDetail = "integra/video/\(argCameraId)"

    static let moduleKey = "integra-video"
    static let titleWall = "Cámaras"
    static let titleDetail = "Cámara"

    static func cameraDetail(_ cameraId: String) -> String {
        let encoded = cameraId.addingPercentEncoding(withAllowedCharacters: .urlPathAllowed) ?? cameraId
        return "integra/video/\(encoded)"
    }

    static func title(for route: String?) -> String? {
        guard let route else { return nil }
        if route == wall { return titleWall }
        if route.hasPrefix("integra/video/") { return titleDetail }
        return nil
    }

    static func route(forModuleKey key: String) -> String? {
        switch key.lowercased() {
        case moduleKey, "video", "cameras", "camaras", "cámaras", "videowall":
            return wall
        default:
            return nil
        }
    }
}
